src/lib/coda_intf/transition_frontier_components_intf.ml

```diff
@@ -233,15 +233,81 @@ module type Transition_handler_intf = sig
      and type transition_frontier_breadcrumb := transition_frontier_breadcrumb
 end
 
+module Best_tip_verification_result = struct
+  type 'external_transition t =
+    {root: 'external_transition; best_tip: 'external_transition}
+end
+
+module type Best_tip_prover_intf = sig
+  type transition_frontier
+
+  type external_transition
+
+  type external_transition_with_initial_validation
+
+  type verifier
+
+  val prove :
+       logger:Logger.t
+    -> transition_frontier
+    -> ( external_transition
+       , State_body_hash.t list * external_transition )
+       Proof_carrying_data.t
+       option
+
+  val verify :
+       verifier:verifier
+    -> ( external_transition
+       , State_body_hash.t list * external_transition )
+       Proof_carrying_data.t
+    -> external_transition_with_initial_validation
+       Best_tip_verification_result.t
+       Deferred.Or_error.t
+end
+
+module type Consensus_best_tip_prover_intf = sig
```

Comment:
Can you add doc-comments here explaining `Best_tip_prover_intf` and `Consensus_best_tip_prover_intf` as well as their differences?